shared/js/https.js

```diff
@@ -1,228 +1,211 @@
-            db.ready().then(() => {
-                this.db = db
+        console.log("HTTPS: init()")
 
-                if (this.isSync) {
-                    this.initSync().then(() => resolve())
-                } else {
-                    resolve()
-                }
+        return new Promise((resolve) => {
+            // Try to load the list from local storage
+            this.getFromStorage().then((items) => {
+
+                // if items were found in local storage,
+                console.log("HTTPS: init() found existing list in storage with " + items.length + " items")
+                httpsUpgradeList = items
+                return resolve()
+
+            }, () => {
+
+                // if it fails to find anything in local storage,
+                // load from the json file packaged with the extension
+                console.log("HTTPS: init() nothing found in storage, loading from local file")
+
+                settings.ready().then(() => {
+                    load.JSONfromLocalFile(constants.httpsUpgradeList, (items) => {
+                        console.log("HTTPS: init() loaded upgrade list from local file with " + items.length + " items")
+                        httpsUpgradeList = items
+                        return resolve()
+                    })
+                })
             })
         })
     }
 
-    initSync () {
-        return new Promise((resolve) => {
-            this.db.logAllRecords(this.dbObjectStore, this.syncRuleCache)
-            resolve()
+    updateList() {
+        console.log("HTTPS: updateList() check if new list exists")
+
+        let etag = settings.getSetting('https-etag') || ''
+        
+        load.loadExtensionFile({
+            url: constants.httpsUpgradeListUrl,
+            source: 'external',
+            etag: etag
+        }, (data, res) => {
+            // this only gets called if the etag is different
+            // and it was able to get a new list
+            console.log("HTTPS: updateList() got updated list from server")
+
+            let newEtag = res.getResponseHeader('etag') || ''
+
+            try {
+                // update list in memory
+                httpsUpgradeList = JSON.parse(data)
+                console.log("HTTPS: updateList() new list has " + httpsUpgradeList.length + " items")
+                
+                this.saveToStorage(data)
+
+                // save new etag for next time
+                settings.updateSetting('https-etag', newEtag)
+                console.log("HTTPS: updateList() updated https-etag to " + newEtag)
+
+            } catch(e) {
+                console.log("HTTPS: updateList() error parsing server response")
+            }
         })
     }
 
-    ready () {
-        return this._ready
-    }
-
-    pipeRequestUrl (reqUrl, tab, isMainFrame) {
+    getFromStorage (fn) {
+        return new Promise((resolve, reject) => {
+            // For Chrome/Firefox:
+            if (window.chrome) {
```

Comment:
Isn't there a util function for this? The same one used previously for setting `isSync`. Why not use that?

Reply:
I suppose that was for isChromeBrowser. In any case, I think it's worth using what we have or changing those.

Reply:
no, this is different. It's doing feature detection, not browser detection. chrome.storage.local is supported by FF and Chrome (and anyone that implements that same extension spec).

Reply:
I'll make the comment clearer here.

Reply:
Alright, is it worth adding that feature detection to utils then and calling that more clearly? Or is this OK and that would be annoyingly redundant? I really don't know.

Reply:
yea, it could be in a util, I'm not sure how Jason is handling it, he probably had to do something similar elsewhere.

Reply:
Fair enough. Bigger fish right now.

Reply:
+1 Would be good to consolidate feature + browser detection + polyfills in one place. Added a task in bug queue to track for later.